stackit/internal/services/loadbalancer/credential/datasource.go

```diff
@@ -0,0 +1,200 @@
+)
+
+type DataSourceModel struct {
+	Id             types.String `tfsdk:"id"` // needed by TF
+	ProjectId      types.String `tfsdk:"project_id"`
+	DisplayName    types.String `tfsdk:"display_name"`
+	Username       types.String `tfsdk:"username"`
+	CredentialsRef types.String `tfsdk:"credentials_ref"`
+}
+
+// NewCredentialDataSource is a helper function to simplify the provider implementation.
+func NewCredentialDataSource() datasource.DataSource {
+	return &credentialDataSource{}
+}
+
+// credentialDataSource is the data source implementation.
+type credentialDataSource struct {
+	client *loadbalancer.APIClient
+}
+
+// Metadata returns the data source type name.
+func (r *credentialDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_loadbalancer_credential"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *credentialDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerData, ok := req.ProviderData.(core.ProviderData)
+	if !ok {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Expected configure type stackit.ProviderData, got %T", req.ProviderData))
+		return
+	}
+
+	var apiClient *loadbalancer.APIClient
+	var err error
+	if providerData.LoadBalancerCustomEndpoint != "" {
+		apiClient, err = loadbalancer.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.LoadBalancerCustomEndpoint),
+		)
+	} else {
+		apiClient, err = loadbalancer.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the data source configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Load balancer client configured")
+}
+
+// Schema defines the schema for the data source.
+func (r *credentialDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":            "Load balancer credential data source schema. Must have a `region` specified in the provider configuration.",
+		"id":              "Terraform's internal resource ID. It is structured as \"`project_id`\",\"`credentials_ref`\".",
+		"project_id":      "STACKIT project ID to which the load balancer credential is associated.",
+		"display_name":    "Credential name.",
+		"username":        "The username used for the ARGUS instance.",
+		"password":        "The password used for the ARGUS instance.",
```

Comment:
There's no password for data source
Does it make sense to have it as a data source?

Reply:
Good point, probably not right? Was in autopilot implementing it here

Reply:
Removed